cdap-cli/src/main/java/co/cask/cdap/cli/command/SendFileStreamCommand.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright © 2015 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.cli.command;
+
+import co.cask.cdap.cli.ArgumentName;
+import co.cask.cdap.cli.CLIConfig;
+import co.cask.cdap.cli.ElementType;
+import co.cask.cdap.cli.util.AbstractAuthCommand;
+import co.cask.cdap.client.StreamClient;
+import co.cask.common.cli.Arguments;
+import com.google.common.collect.Maps;
+import com.google.common.io.Files;
+import com.google.inject.Inject;
+
+import java.io.File;
+import java.io.PrintStream;
+import java.util.Map;
+
+/**
+ * Command for sending file to stream
+ */
+public class SendFileStreamCommand extends AbstractAuthCommand {
+
+  // A map from file extension to content type
+  private static final Map<String, String> CONTENT_TYPE_MAP;
+
+  static {
+    Map<String, String> contentTypes = Maps.newTreeMap(String.CASE_INSENSITIVE_ORDER);
+    contentTypes.put("avro", "avro/binary");
+    contentTypes.put("csv", "text/csv");
+    contentTypes.put("tsv", "text/tsv");
+    contentTypes.put("txt", "text/plain");
+    contentTypes.put("log", "text/plain");
+    CONTENT_TYPE_MAP = contentTypes;
+  }
+
+  private final StreamClient streamClient;
+
+  @Inject
+  public SendFileStreamCommand(StreamClient streamClient, CLIConfig cliConfig) {
+    super(cliConfig);
+    this.streamClient = streamClient;
+  }
+
+  @Override
+  public void perform(Arguments arguments, PrintStream output) throws Exception {
+    String streamId = arguments.get(ArgumentName.STREAM.toString());
+    File file = new File(arguments.get(ArgumentName.LOCAL_FILE_PATH.toString()));
+    String contentType = arguments.get(ArgumentName.CONTENT_TYPE.toString(), "");
+
+    if (contentType.isEmpty()) {
+      contentType = getContentType(Files.getFileExtension(file.getName()));
+    }
+    if (contentType.isEmpty()) {
+      throw new IllegalArgumentException("Unsupported file format.");
+    }
+    if (!file.isFile()) {
```

Comment:
Either way is fine as `getName` doesn't require file to exists. Anyway, moving it up.